rest-service/manager_rest/test/endpoints/test_nodes.py

```diff
@@ -31,146 +31,176 @@ class NodesTest(base_test.BaseServerTestCase):
+        return models.Node(**node_params)
+
+    def _instance(self, instance_id, **kwargs):
+        instance_params = {
+            'id': instance_id,
+            'state': '',
+            'creator': self.user,
+            'tenant': self.tenant,
+        }
+        instance_params.update(kwargs)
+        if 'node' not in instance_params:
+            instance_params['node'] = self._node('node1')
+        return models.NodeInstance(**instance_params)
 
     def test_get_nonexisting_node(self):
         response = self.get('/node-instances/1234')
         self.assertEqual(404, response.status_code)
 
-    def test_get_node(self):
-        put_node_instance(
-            self.sm,
-            instance_id='1234',
-            deployment_id='111',
-            runtime_properties={
-                'key': 'value'
-            }
+    def test_get_instance(self):
+        self._instance(
+            '1234',
+            runtime_properties={'key': 'value'},
         )
         response = self.get('/node-instances/1234')
-        self.assertEqual(200, response.status_code)
-        self.assertEqual('1234', response.json['id'])
-        self.assertTrue('runtime_properties' in response.json)
-        self.assertEqual(1, len(response.json['runtime_properties']))
-        self.assertEqual('value', response.json['runtime_properties']['key'])
+        assert response.status_code == 200
+        assert response.json['id'] == '1234'
+        assert response.json['runtime_properties'] == {'key': 'value'}
 
     def test_sort_nodes_list(self):
-        self.put_deployment(deployment_id='d0', blueprint_id='b0')
-        self.put_deployment(deployment_id='d1', blueprint_id='b1')
+        dep2 = models.Deployment(
+            id='d2',
+            blueprint=self.bp1,
+            scaling_groups={},
+            creator=self.user,
+            tenant=self.tenant,
+        )
+        db.session.add(dep2)
+        node1 = self._node('node1')
+        node2 = self._node('node2', deployment=dep2)
 
         nodes = self.client.nodes.list(sort='deployment_id')
-        for i in range(len(nodes) - 1):
-            self.assertTrue(
-                nodes[i].deployment_id <= nodes[i + 1].deployment_id)
+        assert [n.id for n in nodes] == ['node1', 'node2']
 
-        nodes = self.client.nodes.list(
-            sort='deployment_id', is_descending=True)
-        for i in range(len(nodes) - 1):
-            self.assertTrue(
-                nodes[i].deployment_id >= nodes[i + 1].deployment_id)
+        nodes = self.client.nodes.list(sort='deployment_id',
+                                       is_descending=True)
+        assert [n.id for n in nodes] == ['node2', 'node1']
 
     def test_bad_patch_node(self):
         """Malformed node instance update requests return an error."""
         response = self.patch('/node-instances/1234', 'not a dictionary')
-        self.assertEqual(400, response.status_code)
+        assert response.status_code == 400
```

Comment:
this also rewrites all the `self.assertX` as just `assert`. Not a very important reason for that, but why not.

Tbh, I was mostly using this as a way to see which tests I have already gone through.